### ISSUE TITLE
F.relu(inplace) followed by F.dropout(inplace) breaks backward pass

### ISSUE THREAD

**Issue by andreasveit** [pytorch/pytorch#727]
The following snipped demonstrates the problem:

```
import torch
import torch.nn as nn
from torch.autograd import Variable
import numpy as np
import torch.nn.functional as F

x = Variable(torch.from_numpy(np.random.randn(4, 6, 10,10)).float(), requires_grad=True)
x = F.relu(x, inplace=True)
x = F.dropout(x,p=0.1, inplace=True)
result = torch.sum(x)
result.backward()
```

**Comment by apaszke** [comment]
Yeah, that is expected. You can't apply in-place operations to leaf Variables. Just remove the `inplace` flag and it should be good.
